### DISCUSSION TITLE
FirebaseAuth revokeTokenWithAuthorizationCode() not working

### DISCUSSION THREAD

**Discussion by WellingtonBipo** [firebase/flutterfire#11723] [Q&A] [answered]
When call revokeTokenWithAuthorizationCode(), throws an error.

Error: FirebaseAuthException
code: invalid-credential
message: Code flow is not enabled for Apple.

Following code is generating this issue.

```dart
final provider = AppleAuthProvider();
final credential = await FirebaseAuth.instance.signInWithProvider(provider);
await FirebaseAuth.instance.revokeTokenWithAuthorizationCode(credential.additionalUserInfo!.authorizationCode!);
```

**Comment by dk-dsr** [comment, marked as answer]
Hi @WellingtonBipo ,

I came across the mentioned issue and found a solution that worked for me. In the Firebase console, it's crucial to properly configure the Apple sign-in section. Specifically, you'll need to ensure the "Services ID" and "OAuth code flow configuration" are correctly set up. The UI states that these fields are optional or not required for Apple. However, for token revocation, they are required.

**Reply by WellingtonBipo** [reply]
Hi @dk-dsr. It was that. Worked great. Thanks a lot!!
